ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/receive/view/ReceiveView.kt

```diff
@@ -170,73 +162,139 @@ private fun ReceiveContents(
     versionInfo: VersionInfo,
     modifier: Modifier = Modifier,
 ) {
-    Column(
-        modifier =
-            Modifier
-                .fillMaxHeight()
-                .verticalScroll(rememberScrollState())
-                .then(modifier),
-        horizontalAlignment = Alignment.CenterHorizontally
-    ) {
-        if (screenBrightnessState == ScreenBrightnessState.FULL) {
-            BrightenScreen()
-            DisableScreenTimeout()
+    if (screenBrightnessState == ScreenBrightnessState.FULL) {
+        BrightenScreen()
+        DisableScreenTimeout()
+    }
+
+    val state by remember {
+        derivedStateOf {
+            listOfNotNull(
+                walletAddress.unified,
+                walletAddress.sapling.takeIf { versionInfo.isTestnet },
+                walletAddress.transparent,
+            )
         }
+    }
 
-        Spacer(modifier = Modifier.height(ZcashTheme.dimens.spacingSmall))
+    val pagerState = rememberPagerState { state.size }
 
-        Address(
-            walletAddress = walletAddress.unified,
-            onAddressCopyToClipboard = onAddressCopyToClipboard,
-            onQrImageShare = onQrImageShare,
+    Column(
+        modifier = modifier,
+    ) {
+        AddressTabs(
+            modifier = Modifier.fillMaxWidth(),
+            pagerState = pagerState,
+            tabs = state,
         )
-
-        if (versionInfo.isTestnet) {
-            Spacer(modifier = Modifier.height(ZcashTheme.dimens.spacingHuge))
-
-            Address(
-                walletAddress = walletAddress.sapling,
+        HorizontalPager(
+            modifier = Modifier.fillMaxSize(),
+            state = pagerState,
+            userScrollEnabled = false
+        ) { index ->
+            AddressPage(
+                modifier = Modifier.fillMaxSize(),
+                walletAddress = state[index],
                 onAddressCopyToClipboard = onAddressCopyToClipboard,
                 onQrImageShare = onQrImageShare,
             )
         }
+    }
+}
 
-        Spacer(modifier = Modifier.height(ZcashTheme.dimens.spacingHuge))
-
-        Address(
-            walletAddress = walletAddress.transparent,
-            onAddressCopyToClipboard = onAddressCopyToClipboard,
-            onQrImageShare = onQrImageShare,
-        )
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+private fun AddressTabs(
+    modifier: Modifier,
+    pagerState: PagerState,
+    tabs: List<WalletAddress>,
+    coroutineScope: CoroutineScope = rememberCoroutineScope(),
+    onTabSelected: (index: Int) -> Unit = {},
+) {
+    TabRow(
+        modifier = modifier
+            .padding(horizontal = 64.dp)
```

Comment:
We could use one of the predefined padding sizes from `dimens` to keep paddings as close as possible across different screens. This comment applies to more occurrences below.